lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -1448,25 +1448,45 @@ void ApplyWorkingDir(SmallString &clang_argument, StringRef cur_working_dir) {
   llvm::sys::path::append(clang_argument, cur_working_dir, rel_path);
   llvm::sys::path::remove_dots(clang_argument);
 }
+
+enum ClangArgCategory {
+  ClangArgCategoryNone = 0,
+  ClangArgCategoryUnique = 1 << 0,
+  ClangArgCategoryMacro = 1 << 1,
+};
+
+inline bool IsMultiArgClangFlag(const StringRef& arg) {
+  return arg == "-D" || arg == "-U" || arg == "-working-directory";
+}
+
+ClangArgCategory GetClangArgCategory(const StringRef& arg) {
+  if (arg.size() <= 2 || arg[0] != '-')
+    return ClangArgCategoryNone;
+
+  if (arg[1] == 'D' || arg[1] == 'U')
+    return static_cast<ClangArgCategory>(
+        ClangArgCategoryUnique | ClangArgCategoryMacro);
+
+  return ClangArgCategoryNone;
+}
 } // namespace
 
 void SwiftASTContext::AddExtraClangArgs(std::vector<std::string> ExtraArgs) {
+  swift::ClangImporterOptions &importer_options = GetClangImporterOptions();
```

Comment:
This function has very few dependencies. I previously recommended writing an end-to-end test for the uniquing, but I wonder if we could write a unit test for it. That seems like a better fit.

Reply:
There's a test for `-Werror` skipping in this method, so I was hoping that it's possible to write more tests for logic here :)

Reply:
That would certainly work (we'd need to rename the test to be more generic), I just thought that this is such a nice and small piece of functionality that a unit test is a better fit. In the end, both variants are reasonable.